### PR TITLE
[CSE-70] Fix invalid state of search result views

### PR DESCRIPTION
**Pull request by sectore**
This commit seems to be a bigger one. But don't worry, changes are less functional and more view side (HTML) related.

### CODE REVIEW

File: frontend/src/Explorer/View/Address.purs [hunk 1]
```diff
@@ -29,53 +31,57 @@ addressView state =
...
-                                            , maxPage: length txList
-                                            , changePageAction: AddressPaginateTxs
-                                            , onFocusAction: SelectInputText <<< targetToHTMLInputElement
-                                            }
-                                      ]
-
-                      ]
+                      [ case state ^. currentAddressSummary of
+                            NotAsked  -> emptyAddressTxView
+                            Loading   -> emptyAddressTxView
```
Comment (ksaric) [inline frontend/src/Explorer/View/Address.purs]: Shouldn't this say "Loading..." or something?

File: frontend/src/Explorer/View/Header.purs [hunk 2]
```diff
@@ -23,7 +20,7 @@ headerView state = do
                     [ P.className "logo__container"]
                     [ P.div
                         [ P.className "logo__wrapper"]
-                        [ P.a
+                        [ P.link (toUrl Dashboard)
```
Comment (ksaric) [inline frontend/src/Explorer/View/Header.purs]: Yep. Better.

File: frontend/src/Explorer/View/Address.purs (hunk 1 above)
Comment (sectore) [inline frontend/src/Explorer/View/Address.purs]: @ksaric It's already there within the `addressOverview` located above (in `line 83` https://github.com/input-output-hk/cardano-sl-explorer/blob/fix/CSE-70-search-invalid-state/frontend/src/Explorer/View/Address.purs#L83)

File: frontend/src/Explorer/View/Blocks.purs [hunk 3]
```diff
@@ -86,56 +92,34 @@ emptyBlocksView message =
...
 currentBlocks state =
-    if expanded
-    then slice minBlockIndex (minBlockIndex + maxBlockRows) blocks
-    else slice 0 minBlockRows blocks
+    slice minBlockIndex (minBlockIndex + maxBlockRows) blocks
     where
-        blocks = latestBlocks' state
-        expanded = state ^. dashboardBlocksExpanded
-        currentBlockPage = state ^. (dashboardViewState <<< dashboardBlockPagination)
+        blocks = unwrapLatestBlocks $ state ^. currentBlocksResult
```
Comment (ksaric) [inline frontend/src/Explorer/View/Blocks.purs]: I don't get it. We get the name of the field from state in order to get the field from state?

File: frontend/src/Explorer/View/Dashboard/Blocks.purs [hunk 4]
```diff
@@ -53,3 +59,40 @@ emptyBlocksView message =
...
+        expanded = state ^. dashboardBlocksExpanded
+        currentBlockPage = state ^. (dashboardViewState <<< dbViewBlockPagination)
+        minBlockIndex = (currentBlockPage - 1) * maxBlockRows
+
+blocksFooterView :: State -> P.Html Action
+blocksFooterView state =
+    if expanded then
+        paginationView { label: translate (I18nL.common <<< I18nL.cOf) $ lang'
+                        , currentPage: currentBlockPage
+                        , maxPage: (+) minPagination <<< flip (/) maxBlockRows $ length blocks
```
Comment (ksaric) [inline frontend/src/Explorer/View/Dashboard/Blocks.purs]: Simpler is `(minPagination / maxBlockRows) + (length blocks)` if that's the goal.
This is getting repetitive, maybe it's time we build a simple `purescript-table` component?

File: frontend/src/Explorer/View/Blocks.purs [hunk 5]
```diff
@@ -59,11 +57,19 @@ blocksView state =
...
+                , case state ^. currentBlocksResult of
                       NotAsked  -> emptyBlocksView ""
                       Loading   -> emptyBlocksView $ translate (I18nL.common <<< I18nL.cLoading) lang'
-                      Failure _ -> emptyBlocksView $ translate (I18nL.block <<< I18nL.blNotFound) lang'
+                      Failure _ -> failureView lang'
                       Success blocks ->
+                          let paginationViewProps =
+                                  { label: translate (I18nL.common <<< I18nL.cOf) $ lang'
+                                  , currentPage: state ^. (viewStates <<< blocksViewState <<< blsViewPagination)
+                                  , maxPage: (+) minPagination <<< flip (/) maxBlockRows $ length blocks
```
Comment (ksaric) [inline frontend/src/Explorer/View/Blocks.purs]: Simpler is `(minPagination / maxBlockRows) + (length blocks)` if that's the goal.

File: frontend/src/Explorer/View/Address.purs (hunk 1 above)
Comment (ksaric) [inline frontend/src/Explorer/View/Address.purs]: @sectore Oh, ok, didn't notice that.

**Comment by ksaric** [comment]
I didn't forget about `RemoteData` refinement, but didn't have much time, I'll try to create an example over the week.

File: frontend/src/Explorer/View/Blocks.purs (hunk 5 above)
Comment (sectore) [inline frontend/src/Explorer/View/Blocks.purs]: @ksaric The goal is: `minPagination + ((length blocks) / maxBlockRows)`

Comment (ksaric) [inline frontend/src/Explorer/View/Blocks.purs]: @sectore Ok, that's still simpler.

File: frontend/src/Explorer/View/Blocks.purs (hunk 3 above)
Comment (sectore) [inline frontend/src/Explorer/View/Blocks.purs]: @ksaric `unwrapLatestBlocks` is a just helper function for views only. It unwraps the `Success` value or returns an empty `[]` for all the other cases (see https://github.com/input-output-hk/cardano-sl-explorer/blob/fix/CSE-70-search-invalid-state/frontend/src/Explorer/View/Blocks.purs#L109-L113)

File: frontend/src/Explorer/View/Dashboard/Blocks.purs (hunk 4 above)
Comment (sectore) [inline frontend/src/Explorer/View/Dashboard/Blocks.purs]: @ksaric See https://github.com/input-output-hk/cardano-sl-explorer/pull/43#discussion_r111397035

File: frontend/src/Explorer/View/Blocks.purs (hunk 3 above)
Comment (ksaric) [inline frontend/src/Explorer/View/Blocks.purs]: @sectore That's ok, the part I don't get is `state ^. currentBlocksResult` and `currentBlockPage = state ^. (viewStates <<< blocksViewState <<< blsViewPagination)`? That's screaming to me something is wrong?

Comment (sectore) [inline frontend/src/Explorer/View/Blocks.purs]: @ksaric What is wrong with this? Have you checked the types of these values?
- `currentBlocksResult` https://github.com/input-output-hk/cardano-sl-explorer/blob/fix/CSE-70-search-invalid-state/frontend/src/Explorer/Types/State.purs#L37
- `blsViewPagination` https://github.com/input-output-hk/cardano-sl-explorer/blob/fix/CSE-70-search-invalid-state/frontend/src/Explorer/Types/State.purs#L102

Comment (ksaric) [inline frontend/src/Explorer/View/Blocks.purs]: @sectore All good. I misread `currentBlocksResult` and thought it was `currentBlockPage`.

Comment (sectore) [inline frontend/src/Explorer/View/Blocks.purs]: @ksaric Ok. Maybe it is not the best naming of these properties. We can rename anything if there is any confusion.

File: frontend/src/Explorer/View/Dashboard/Blocks.purs (hunk 4 above)
Comment (sectore) [inline frontend/src/Explorer/View/Dashboard/Blocks.purs]: @ksaric The `paginationView` is already a component, which has no dependencies to state, it takes just properties https://github.com/input-output-hk/cardano-sl-explorer/blob/fix/CSE-70-search-invalid-state/frontend/src/Explorer/View/Common.purs#L201-L266 

The same for table header (`txHeaderView`) and table body (`txBodyView`):
- `txBodyView` https://github.com/input-output-hk/cardano-sl-explorer/blob/fix/CSE-70-search-invalid-state/frontend/src/Explorer/View/Common.purs#L141-L154
- `txHeaderView` https://github.com/input-output-hk/cardano-sl-explorer/blob/fix/CSE-70-search-invalid-state/frontend/src/Explorer/View/Common.purs#L83-L104

File: frontend/src/Explorer/View/Common.purs [hunk 6]
```diff
@@ -265,6 +266,11 @@ paginationView props =
               then props.changePageAction value
               else NoOp
 
+
+getMaxPaginationNumber :: Int -> Int -> Int
+getMaxPaginationNumber quantity max =
```
Comment (ksaric) [inline frontend/src/Explorer/View/Common.purs]: Good idea.

**Pull request merged**